### PR TITLE
[Bug] Use torch.no_grad() in the QNG-SPSA algorithm

### PR DESCRIPTION
**Pull request by inafergra**
We were accidentally tracking the gradients when calculating the QFI matrix and updating the model's parameters in the QNG-SPSA algorithm, which slows down the calculations and uses up a lot of memory

### CODE REVIEW

**Comment by jpmoutinho** [comment]
Thanks @inafergra, is this ready for review?

**Comment by inafergra** [comment]
> Thanks @inafergra, is this ready for review?

Yes! Forgot to change from draft

**Review by jpmoutinho — APPROVED** [review]
Well LGTM, not much to say :) I assume you ran some benchmark to check this does indeed fix the issue.
